integration/experiment/ffnet/README.md

```diff
@@ -0,0 +1,243 @@
+                           GPUs on the node.
+
+  - `--step-gpu-frequency`: the step size in hertz between settings for the sweep.
+                            By default, it uses the minimum supported step size
+                            of GPUs on the node.
+
+  Note that for any frequency domain, if the values above are not specified or if
+  the min and max values are equal, a frequency sweep will not be executed across
+  that domain.
+
+## Supporting Files
+
+#### `gen_hdf_from_fsweep.py`:
+
+  Generates HDF files used to generate the neural net and frequency recommendation json files.
+  This takes report and trace files from frequency sweep experiments as inputs, checks for
+  required signals on CPU and GPU domains, annotates data with microbenchmark and node names,
+  and outputs the HDFs. Two files are generated: The stats file contains per-region report
+  information used to determine frequency recommendation for each region class. The trace file
+  contains trace data that is used to create a neural net that determines region class 
+  probabilities during a workload execution.
+
+  This script requires the following positional inputs:
+
+  - `output`: Prefix for output files `[output]_stats.h5` and `[output]_traces.h5`
+  - `frequency_sweep_dirs`: Directories containing reports and traces from frequency sweeps
+    
+  E.g.:
+
+   ```
+   ./gen_hdf_from_fsweep.py example_output /path/to/fsweep
+   ```
+   This will generate `example_output_stats.h5` and `example_output_traces.h5`
+
+#### `gen_neural_net.py`:
+
+   Generates neural net json file(s) for CPU and/or GPU. This takes in the trace h5 file generated 
+   from `gen_hdf_from_fsweep.py` which is annotated with region classes. Required signals are 
+   specified within this script, per-domain. The script checks for the complete list of signals
+   and generates the corresponding neural nets.
+
+   This script can take in the following inputs:
+
+   - `output`: Prefix of the output json file(s).
+   - `description`: Description of the neural net
+   - `ignore` : A comma-separated list of region hashes to ignore
+   - `data` : Data files to train on. This can take in multiple trace HDF files.
+
+   Note that this script depends upon pytorch. Pytorch can be installed using: `pip install pytorch`
+  
+   E.g.:
+
+   ```
+   ./gen_neural_net.py --output nnet --description "A neural net" --data example_output_traces.h5
+   ```
+   This will generate `nnet_cpu.json` and/or `nnet_gpu.json`
+
+#### `gen_region_parameters.py`:
+
+   Generates region-class frequency recommendation map. This takes in the stats h5 file
+   generated from `gen_hdf_from_fsweep.py`, which is annotated with region classes and
+   contains runtime and per-domain energy information. This is used to generate a
+   runtime-frequency relationship and determine minimum energy point within an allowable
+   perf degradation. For each region class, a list is generated which contains the 
```

Comment:
Minor nit:

```suggestion
   performance degradation. For each region class, a list is generated which contains the 
```

I checked, and we're not using the shorthand "perf" anywhere else in our docs.